udata/harvest/tests/test_api.py

```diff
@@ -307,11 +307,31 @@ def test_update_source(self, api):
         }
         api_url = url_for('api.harvest_source', ident=str(source.id))
         response = api.put(api_url, data)
+        assert200(response)
+        assert response.json['url'] == new_url
 
+        # Source is now owned by orga, with user as member
+        source.organization = OrganizationFactory(members=[Member(user=user)])
+        source.save()
+        api_url = url_for('api.harvest_source', ident=str(source.id))
+        response = api.put(api_url, data)
         assert200(response)
 
-        source = response.json
-        assert source['url'] == new_url
+    def test_update_source_require_permission(self, api):
+        '''It should not update a source if not the owner'''
+        api.login()
+        source = HarvestSourceFactory()
+        new_url = faker.url()
```

Comment:
I would add a little type hint here so that reader knows what's the type without having to inspect `faker.url()` method, but might be because I'm a type hint nazi :)